src/lib/components/inventory/selected_item_info.ts

```diff
@@ -203,6 +230,21 @@ export class SelectedItemInfo extends FloatElement {
             } catch (e: any) {
                 console.error(`Failed to fetch float for ${this.asset.assetid}: ${e.toString()}`);
             }
+
+            // Fetch bluegem data if needed
+            if (this.itemInfo && isBlueSkin(this.itemInfo)) {
```

Comment:
We should probably reset the value of `bluegemData` if this is false (or we fail to fetch below).

Otherwise, I think switching between two blue gems, where the second one fails to fetch below, would show incorrect previous blue gem data.